doc/decisions/1_in_discussion/internal_cache.md

```diff
@@ -136,6 +136,46 @@ ksRemoveByName (cowMeta, "meta:/type");
 
 - Elektra doesn't require MMAP
 
+**Cons:**
+
+- Lifetime of a copied COW key MUST be less than the key it was copied from.
+  We can not track how many keys point to the same data this way, so we can only free data if the key does not have the COW flag.
+  If the original key gets deleted, using a COW key that points to the same data will lead to corrupt data.
+  The same is true for updating values of the original key.
+  
+  This is only problematic if we want to use COW for keys outside of `KDB`.
+  If it is only for use within `KDB`, especially for usage as internal cache and in change tracking, we always know that the original keys are going to last as long as the `KDB` instance.
```

Comment:
```suggestion
  If it is only for use within `KDB`, especially for usage as internal cache and in change tracking, we could always guarantee that the original keys are going to last as long as the `KDB` instance.
```

It is not automatically guaranteed that keys last until `kdbClose`. For change tracking it would probably make much more sense if the keys were only valid until the next `kdbGet`/`kdbSet`. Otherwise we would have to keep all versions of keys internally until `kdbClose`.

Also I just checked and it turns out that data returned by `mmapstorage` is only valid until the next `kdbSet`, because that's when `munmap` is called. But I think (at least for the cache) we could move that to `close`.

Sorry for the confusion because of my earlier comments, should have checked the code first.